Frameworks/VMFleet/VMFleet.psm1

```diff
@@ -2263,259 +2440,335 @@ function New-Fleet
+                                            LogOutput "Result returned for $name - $vmResult"
+                                            if($vmResult -eq $null){
+                                                $msg = "Error creating Virtual machine $name at retry count $retryCount"
+                                                LogOutput -ForegroundColor Yellow $msg
+                                                throw $msg
+                                            }
+                                            break
+                                       }
+                                       catch {
+                                          $retrycount++
+                                          if($retrycount -eq 3){
+                                               LogOutput -ForegroundColor Yellow "Error occurred while creating $name for all retry counts"
+                                               $stop = $true
+                                          }
+                                       }
+                                   }
+                                   if (-not $stop) {
+                                       # validating creation of Arc VMs using powershell commands
+                                       # hyperv Arc VM names
+                                       $vmname = "Virtual Machine " + $name
+                                       $Stoploop = $false
+                                       $Retrycount = 0
+                                       do {
+                                           try {
+                                               LogOutput "Executing Get-ClusterResource -Name $vmname"
+                                               $o = Get-ClusterResource -Name $vmname
+                                               if ($o -eq $null) {
+                                                   throw "Null returned by Get-ClusterResource due to some error"
+                                               }
+                                               else {
+                                                  LogOutput "Creation of vm completed at hyperv level"
+                                                   $Stoploop = $true
+                                               }
+                                           }
+                                           catch {
+                                               if ($Retrycount -gt 10) {
+                                                   LogOutput "Could not get VM $vmname after 10 retries using Get-ClusterResource"
+                                                   $Stoploop = $true
+                                               }
+                                               else {
+                                                   Write-Host "Could not get VM $vmname at $RetryCount retry count. Retrying in 6 minutes..."
```

Comment:
consider LogOutput so it comes with a timestamp? note the other branch of the if immediately above